modules/olm-enabling-operator-restricted-network.adoc

```diff
@@ -18,18 +19,140 @@ As an Operator author, your Operator must meet additional requirements to run pr
+----
+====
+
+. Pull the related images with digests by replacing hard-coded image references with environment variables in the following files:
+
+* For Go-based Operator projects, update the `controllers/memcached_controller.go` file with the following changes:
++
+.`controllers/memcached_controller.go` diff
+[%collapsible]
+====
+[source,diff]
+----
+...
+// deploymentForMemcached returns a memcached Deployment object
+...
+	Spec: corev1.PodSpec{
+        	Containers: []corev1.Container{{
+-			Image:   "memcached:1.4.36-alpine",
++			Image:   os.Getenv("RELATED_IMAGE_MEMCACHED"),
+			Name:    "memcached",
+			Command: []string{"memcached", "-m=64", "-o", "modern", "-v"},
+			Ports: []corev1.ContainerPort{{
+...
+----
++
+The `os.Getenv` function returns an empty string in the variable is not set. Set the environment variable before changing the file.
+====
 
-.. Update `spec.relatedImages`:
+* For Ansible-based Operator projects, update the `roles/memcached/tasks/main.yml` file with the following changes:
 +
+.`roles/memcached/tasks/main.yml` diff
+[%collapsible]
+====
+[source,diff]
+----
+...
+spec:
+  containers:
+  - name: memcached
+    command:
+    - memcached
+    - -m=64
+    - -o
+    - modern
+    - -v
+-   image: "docker.io/memcached:1.4.36-alpine"
++   image: "{{ lookup('env', 'RELATED_IMAGE_MEMCACHED') }}"
+    ports:
+      - containerPort: 11211
+...
+----
+====
+
+* For Helm-based Operator projects, update the `helm-charts/memchached/values.yaml` file with the following changes:
++
+.`helm-charts/memchached/values.yaml` diff
+[%collapsible]
+====
+[source,diff]
+----
+  ## Memcached image and tag
+  ## ref: https://hub.docker.com/r/library/memcached/tags/
+  ##
+- image: memcached:1.5.20
++ image: "{{ lookup('env', 'RELATED_IMAGE_MEMCACHED') }}"
+...
+----
+====
+
+. Add the `BUNDLE_GEN_FLAGS` variable definition to your `Makefile` file and make the following changes:
++
+.`Makefile` diff
+[source,diff]
+----
+...
+BUNDLE_GEN_FLAGS ?= -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
+
+# USE_IMAGE_DIGESTS defines if images are resolved via tags or digests
+# You can enable this value if you would like to use SHA Based Digests
+# To enable set flag to true
+USE_IMAGE_DIGESTS ?= false
+ifeq ($(USE_IMAGE_DIGESTS), true)
+	BUNDLE_GEN_FLAGS += --use-image-digests
+endif
+
+-  $(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
++  $(KUSTOMIZE) build config/manifests | operator-sdk generate bundle $(BUNDLE_GEN_FLAGS)
+...
+----
+
+. Use the `make bundle` command and set `USE_IMAGE_DIGESTS` to `true` to automatically update your Operator image to use a digest (SHA) and not a tag:
++
+[source,terminal]
+----
+$ make bundle USE_IMAGE_DIGESTS=true
+----
++
+[TIP]
+====
+
+Alternatively, you can update your Operator project manually by using SHA references to related images in two places in the CSV for your Operator:
+
+* Update `spec.relatedImages`:
```

Comment:
Just a thought: I am wondering if it would be better for these to be numbered instead of bullets since you have "two places" in the initial sentence.

Reply:
or if you want to leave the bullet, you could say something like "The first place you can update ..."